src/store/modules/order/actions.ts

```diff
@@ -312,6 +312,20 @@ const actions: ActionTree<OrderState , RootState> ={
     await dispatch('fetchShipGroupForOrder');
   },
 
+  async updateOrderItemFetchingStatus ({ commit, state }, payload) {
+    const order = state.current ? JSON.parse(JSON.stringify(state.current)) : {};
+
+    order.shipGroups?.find((shipGroup: any) => {
+      if(shipGroup.shipGroupSeqId === payload.shipGroupSeqId){
+        shipGroup.items?.find((item: any) => {
+          if(item.productId === payload.productId) item.isFetchingStock = !item.isFetchingStock
```

Comment:
Instead of adding this whole logic, we can simply update the item.isFetchingStock in the component method directly and then directly update the current order.